packages/mdx/mdx-ast-to-mdx-hast.js

```diff
@@ -1,102 +1,80 @@
-const toHAST = require('mdast-util-to-hast')
+const toHast = require('mdast-util-to-hast')
 const all = require('mdast-util-to-hast/lib/all')
 const detab = require('detab')
 const u = require('unist-builder')
 
 function mdxAstToMdxHast() {
-  return (tree, _file) => {
-    const handlers = {
-      // `inlineCode` gets passed as `code` by the HAST transform.
-      // This makes sure it ends up being `inlineCode`
-      inlineCode(h, node) {
-        return Object.assign({}, node, {
-          type: 'element',
-          tagName: 'inlineCode',
-          properties: {},
-          children: [
-            {
-              type: 'text',
-              value: node.value
-            }
-          ]
-        })
-      },
-      code(h, node) {
-        const value = node.value ? detab(node.value + '\n') : ''
-        const lang = node.lang
-        const props = {}
+  return tree => {
+    return toHast(tree, {
+      handlers: {
+        // `inlineCode` gets passed as `code` by the HAST transform.
+        // This makes sure it ends up being `inlineCode`
+        inlineCode(h, node) {
+          return h(node, 'inlineCode', [{type: 'text', value: node.value}])
+        },
+        code(h, node) {
+          const value = node.value ? detab(node.value + '\n') : ''
+          const lang = node.lang
+          const props = {}
 
-        if (lang) {
-          props.className = ['language-' + lang]
-        }
+          if (lang) {
+            props.className = ['language-' + lang]
+          }
+
+          props.metastring = node.meta
 
-        // MDAST sets `node.meta` to `null` instead of `undefined` if
-        // not present, which React doesn't like.
-        props.metastring = node.meta || undefined
+          // To do: this handling seems not what users expect:
+          // <https://github.com/mdx-js/mdx/issues/702>.
+          const meta =
+            node.meta &&
+            node.meta.split(' ').reduce((acc, cur) => {
+              if (cur.split('=').length > 1) {
+                const t = cur.split('=')
+                acc[t[0]] = t[1]
+                return acc
+              }
```

Comment:
I'm a bit surprised there is a custom meta handler here.
Since there is so much variation in what user's want/expect, it may make sense to split it out into a plugin?

Reply:
All for it. I added a comment above this to link to an issue (and there's another one open as well). This current behavior has some value, but the implementation is not what folks seem to expect. I left it as is to not break anything, but would like to look into it again